controllers/odk.py

```diff
@@ -1,31 +1,40 @@
 import io
-from flask import Blueprint, jsonify, request, make_response, g
+from flask import Blueprint, jsonify, request, make_response
 from models.odk import Odk
-from models.user import User
 from odk2odm import odk_requests
+from flask_security import login_required, current_user
 
-from flask_httpauth import HTTPBasicAuth
-auth = HTTPBasicAuth()
 # API components that retrieve or download data from database for use on front end
 odk_api = Blueprint("odk_api", __name__)
 
-@auth.verify_password
-def verify_password(username, password):
-    user = User.query.filter_by(email=username).first()
-    if not user or not user.check_password(password):
-        return False
-    g.user = user
-    return True
+def get_odk(id):
+    """
+    Retrieve config with id from database, check if it belongs to user
+    :param id: odm id
+    :return: if the chosen project does not belong to the user, then return a 403 response, else return the Odm model
+    """
+    try:
+        return Odk.query.filter_by(id=id).filter(Odk.user_id == current_user.id).one()
+    except:
+        return jsonify("Access denied"), 403
+
+@odk_api.before_request
+def before_request():
+    if current_user.is_anonymous:
+        return jsonify("Forbidden"), 401
 
 @odk_api.route("/api/odk/<id>/users", methods=["GET"])
+@login_required
 def get_users(id):
     """
     API endpoint for getting list of users on ODK Central server
 
     :param id: int - ODK server id
     :return: http response
     """
-    odk = Odk.query.get(id)
+    odk = get_odk(id)
+    if not(isinstance(odk, Odk)):
```

Comment:
In what circumstances would this not be an instance of ODK? If the user wasn't properly authenticated? Or if something else went wrong?